### PR TITLE
Update Twitter name to X in Rails release documentation

### LINKED ISSUE #52510: Twitter to X update
### Expected behavior

The Rails release documentation should reflect the current and correct social media account name.

### Actual behavior

The Rails release documentation currently mentions the old Twitter account name instead of the new name "X".

### PR DESCRIPTION
**Pull request by SarthakPaandey**
### Motivation / Background

This Pull Request updates the Twitter account name in the Rails release documentation to the new name "X" to ensure that all references are current and correct.

### Detail

This Pull Request changes the Twitter account name mentioned in the Rails release documentation from the old name to "X".

### Additional information

This change is straightforward and does not require additional context or reference as it's a direct update of social media information.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] [Fix #52510 ]
* [x] Tests are not required for this change as it is a minor documentation update.
* [x] CHANGELOG files do not need an update for this change as it is minor and does not affect the functionality or the API.

### CODE REVIEW

File: guides/source/5_0_release_notes.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ information.
 ### API Applications
 
 Rails can now be used to create slimmed down API only applications.
-This is useful for creating and serving APIs similar to [Twitter](https://dev.twitter.com) or [GitHub](https://developer.github.com) API,
+This is useful for creating and serving APIs similar to [X](https://dev.x.com) or [GitHub](https://developer.github.com) API,
```
Comment (vipulnsward) [inline guides/source/5_0_release_notes.md:58]: https://developer.x.com/

**Comment by fatkodima** [comment]
I think it will always be `Twitter` in our hearts.

File: guides/source/5_0_release_notes.md (hunk 1 above)
Comment (SarthakPaandey) [reply guides/source/5_0_release_notes.md:58]: Thanks for pointing out. Changing right now.

**Comment by fatkodima** [comment]
One of the duplicate PRs can be closed. And please keep the PR changes in 1 commit.

**Pull request closed without merging**